design-proposals/virt-operator-deploy-instancetypes.md

```diff
@@ -0,0 +1,97 @@
+# Overview
+
+The [`common-instancetypes`](https://github.com/kubevirt/common-instancetypes)
+project provides a set of useful [instance types and preferences](https://kubevirt.io/user-guide/virtual_machines/instancetypes/) for
+use when creating
+[`VirtualMachines`](http://kubevirt.io/api-reference/main/definitions.html#_v1_virtualmachine)
+within KubeVirt. Until now operators and users could manually deploy these
+resources from the [upstream project
+tree](https://github.com/kubevirt/common-instancetypes#installation), from a
+[specific release](https://github.com/kubevirt/common-instancetypes/releases/tag/v0.3.1)
+or by using the [SSP operator](https://github.com/kubevirt/ssp-operator#ssp-operator).
+
+This proposal suggests moving the deployment of the cluster-wide
+`common-instancetypes` resources to `virt-operator` so that they are
+consistently available across KubeVirt deployments.
+
+## Motivation
+
+The ultimate goal of this proposal is to improve KubeVirt's default user
+experience when creating `VirtualMachines` across several different deployments.
+
+The deployment by default of the cluster-wide `common-instancetypes` resources
+will allow users to rely more freely on the presence of the resources and make
+use of them during their `VirtualMachine` creation flows.
+
+## Goals
+
+* For new deployments, the deployment of the cluster-wide `common-instancetypes`
+  resources by `virt-operator`
+
+* For upgraded deployments, to have `virt-operator` claim ownership of any
+  existing cluster-wide `common-instancetypes` resources deployed by the SSP
+  operator.
+
+## Non Goals
+
+* For upgraded deployments, `virt-operator` will not try to claim ownership of
+  any operator deployed cluster-wide `common-instancetypes` resources.
+
+## User Stories
+
+* As a user I want access to a consistent default set of cluster-wide instance
+  types and preferences across all deployments I interact with
+
+* As an operator I want my users to have access to a consistent set of
+  cluster-wide instance types and preferences without the need for additional
+  deployment of operators or resources
+
+## Repos
+
+* kubevirt/kubevirt
+* kubevirt/common-instancetypes
+* kubevirt/ssp-operator
+
+# Design
+
+**TODO** Add more context here once some draft code has been written
+
+`virt-operator` will be extended to by default deploy bundles of cluster-wide
+resources from the `common-instancetypes` project.
```

Comment:
@davidvossel just following up, would a git sub-module based approach be acceptable or do you want the full tree imported?

Reply:
I'm not sure if a submodule approach will fly. Mainly because /api is a reverse-submodule approach (we export from kubevirt/kubevirt to a dedicate repo).

The biggest benefit of having (some) ITs+prefs within kubevirt/kuevirt is testing.
But to recude the impact, we should slim down the default set of ITs+Prefs, this would then leave us with a "supplemental" set of ITs+Prefs.

FWIW if we inlcude them in kubevirt/kubevirt, then they should be really flat. To avoid adding to much magic to this repository.

Reply:
> I'm not sure if a submodule approach will fly. Mainly because /api is a reverse-submodule approach (we export from kubevirt/kubevirt to a dedicate repo).

I don't really follow what the objection would be to the submodule approach then given what we do with `kubevirt/api`?

> The biggest benefit of having (some) ITs+prefs within kubevirt/kuevirt is testing. But to recude the impact, we should slim down the default set of ITs+Prefs, this would then leave us with a "supplemental" set of ITs+Prefs.

I think you mean the biggest drawback right? Having the full tree imported into `kubevirt/kubevirt` moves the burden of testing, bug and PR tracking into that repo.

> FWIW if we inlcude them in kubevirt/kubevirt, then they should be really flat. To avoid adding to much magic to this repository.

I'd rather keep the projects separated, I still think the structure of the `kubevirt/common-instancetypes` repo would be useful for certain end users looking to build on the resources we provide there.